scdrs/pp.py

```diff
@@ -3,6 +3,48 @@
 from scipy import sparse
 import pandas as pd
 from skmisc.loess import loess
+from typing import List
+
+
+def category2dummy(
+    df: pd.DataFrame, cols: List[str] = None, verbose: bool = False
+) -> pd.DataFrame:
+    """
+    Convert categorical variables in a dataframe to binary dummy variables.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Dataframe to convert
+    cols : List[str], optional
+        Columns to convert, by default None (columns are then selected automatically)
+
+    Returns
+    -------
+    pd.DataFrame
+        Converted dataframe
+    """
+    df = df.copy()
+
+    if cols is None:
+        cols = list(set(df.columns) - set(df._get_numeric_data().columns))
+
+    added_cols = []
+    for col in cols:
+        # create study dummies variables
+        dummies = pd.get_dummies(df[col], drop_first=True)
+        dummies.columns = [f"{col}_{s}" for s in dummies.columns]
+        dummies.loc[df[col].isnull(), dummies.columns] = np.nan
+        added_cols.extend(dummies.columns)
+        df = pd.concat([df, dummies], axis=1)
```

Comment:
Probably not important. I just learned that frequent pd.concat is inefficient. A better practice is to save the temp dfs in a list and do only one concat at the end. https://stackoverflow.com/questions/36489576/why-does-concatenation-of-dataframes-get-exponentially-slower 

Not important here because df_cov is not a large file.